src/services/security/modules/RecipientAddressModule/index.test.ts

```diff
@@ -0,0 +1,1113 @@
+import * as sdk from '@safe-global/safe-gateway-typescript-sdk'
+import { OperationType } from '@safe-global/safe-core-sdk-types'
+import { ZERO_ADDRESS } from '@safe-global/safe-core-sdk/dist/src/utils/constants'
+import { ethers } from 'ethers'
+import { hexZeroPad } from 'ethers/lib/utils'
+import type { JsonRpcProvider } from '@ethersproject/providers'
+import type { SafeInfo } from '@safe-global/safe-gateway-typescript-sdk'
+import type { SafeTransaction } from '@safe-global/safe-core-sdk-types'
+
+import * as web3 from '@/hooks/wallets/web3'
+import { ERC20__factory, ERC721__factory, Multi_send__factory } from '@/types/contracts'
+import { RecipientAddressModule } from '.'
+
+const getMockErc20TransferCalldata = (to: string) => {
+  const erc20Interface = ERC20__factory.createInterface()
+  return erc20Interface.encodeFunctionData('transfer', [
+    to,
+    0, // value
+  ])
+}
+
+const getMockErc721TransferFromCalldata = (to: string) => {
+  const erc721Interface = ERC721__factory.createInterface()
+  return erc721Interface.encodeFunctionData('transferFrom', [
+    ZERO_ADDRESS, // from
+    to,
+    0, // value
+  ])
+}
+
+const getMockErc721SafeTransferFromCalldata = (to: string) => {
+  const erc721Interface = ERC721__factory.createInterface()
+  return erc721Interface.encodeFunctionData('safeTransferFrom(address,address,uint256)', [
+    ZERO_ADDRESS, // from
+    to,
+    0, // value
+  ])
+}
+
+const getMockErc721SafeTransferFromWithBytesCalldata = (to: string) => {
+  const erc721Interface = ERC721__factory.createInterface()
+  return erc721Interface.encodeFunctionData('safeTransferFrom(address,address,uint256,bytes)', [
+    ZERO_ADDRESS, // from
+    to,
+    0, // value
+    '0x', // bytes
+  ])
+}
+
+const getMockMultiSendCalldata = (recipients: Array<string>): string => {
+  // MultiSendCallOnly
+  const OPERATION = 0
+
+  const data = '0x'
+
+  const internalTransactions = recipients.map((recipient) => {
```

Comment:
The safe-core-sdk exports a util function  `encodeMultiSendData` for this.

Reply:
I had/am having issues with using that helper. As this is purely testing related, I think it's alright keeping it as is. What do you think?